### PR TITLE
Change the word order of some of the toast messages

### PR DESCRIPTION
**Pull request by snowtimeglass**
## Purpose / Description
Anki Desktop uses the following word order pattern on some toast messages, for example, `Card suspended.`,

1. Subject
2. omission of be Verb
3. Past participle

but it isn't used on some strings of AnkiDroid, for example, `Suspended card`.
It is probably appropriate to use Anki Desktop's word order pattern on AnkiDroid as well.

image

## Fixes
n/a

## Approach
Change the strings on 02-strings.xml

## How Has This Been Tested?
on my physical device
imageimageimage
imageimage
image

## Learning (optional, can help others)
n/a

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by BrayanDSO** [comment]
Actually, we have access to Anki Desktop strings, so we can reuse them directly. I'm on my phone right now, so I can't link the exact implementation on the moment, but if I recall correctly, it is on `col.TR`. So that is an option.

As the current PR only changes the English translation, I'd approve it, but I would like to see if the approach of reusing the Anki Desktop strings directly wouldn't be better

**Comment by mikehardy** [comment]
Indeed - here's an example usage https://github.com/ankidroid/Anki-Android/pull/11740/files#diff-03de22cd624f1cb4006ae0c6622ec035f598d779dd1dfb64e2207f0a1824933cR42

And I just logged a couple items with regard to backend translation - it's all pretty new for us (#12452 + https://github.com/ankidroid/Anki-Android-Backend/issues/235)

**Comment by dae** [comment]
You can also use CollectionManager.TR(), which doesn't require an open collection or usage of withCol

**Comment by dae** [comment]
Here's how you could do it:

```diff
diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt b/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt
index a13e4e352..c122b8bd8 100644
--- a/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt
@@ -50,6 +50,7 @@ import com.drakeet.drawer.FullDraggableContainer
 import com.google.android.material.snackbar.Snackbar
 import com.ichi2.anim.ActivityTransitionAnimation
 import com.ichi2.anim.ActivityTransitionAnimation.getInverseTransition
+import com.ichi2.anki.CollectionManager.TR
 import com.ichi2.anki.CollectionManager.withCol
 import com.ichi2.anki.UIUtils.showThemedToast
 import com.ichi2.anki.cardviewer.*
@@ -908,6 +909,15 @@ abstract class AbstractFlashcardViewer :
         }
     }
 
+    private fun showSnackbarWithUndoButtonText(
+        text: String,
+        duration: Int = Snackbar.LENGTH_SHORT
+    ) {
+        showSnackbarAboveAnswerButtons(text, duration) {
+            setAction(R.string.undo) { undo() }
+        }
+    }
+
     private fun getRecommendedEase(easy: Boolean): Int {
         return try {
             when (answerButtonCount) {
@@ -1648,7 +1658,7 @@ abstract class AbstractFlashcardViewer :
 
     internal fun buryCard(): Boolean {
         return dismiss(BuryCard(currentCard!!)) {
-            showSnackbarWithUndoButton(R.string.buried_card)
+            showSnackbarWithUndoButtonText(TR.studyingCardsBuried(1))
         }
     }
 
```

**Comment by snowtimeglass** [comment]
Thank you all for the information and advice. I'll try :-D

**Comment by snowtimeglass** [comment]
>I would like to see if the approach of reusing the Anki Desktop strings directly wouldn't be better

@BrayanDSO I've tried the approach with `CollectionManager.TR()` as @dae taught me.
image

My concern is that the approach makes inconsistency about punctuation between the reused messages and the other ones, at least for the moment. In general, the messages of AnkiDroid end _without_ period as follows
image
while ones of Anki Desktop generally (maybe) end with it.
image
image

**Comment by mikehardy** [comment]
Using Anki translations is something I prefer as it represents a more "shared work" future (between Anki Desktop and AnkiDroid), which may also be read as "less work per contributor" and is thus more sustainable long-term, even if there is punctuation inconsistency.

**Comment by BrayanDSO** [comment]
Anyone can remove the dot on Poonton (Anki desktop translating platform) later. Sharing translations is the best path for consistency

**Comment by snowtimeglass** [comment]
I see. Then, I'll continue this approach without so much concern about temporary inconsistency of punctuation.

By the way, I have one question.
> ```diff
> diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt b/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt
> -            showSnackbarWithUndoButton(R.string.buried_card)
> +            showSnackbarWithUndoButtonText(TR.studyingCardsBuried(1))

For example, if the above replacement is done, should I delete 
`<string name="buried_card">Card buried</string>` 
in [AnkiDroid/src/main/res/values/02-strings.xml](https://github.com/ankidroid/Anki-Android/compare/main...snowtimeglass:Anki-Android:word_order#diff-55582f3477f88d22e8e7acfbac5e2fdea81e4f8f298e0b1b91ec60093e6e3a04)?

**Comment by BrayanDSO** [comment]
If the string isn't somewhere else, you can delete it. Actually, Lint will force will to delete it on this case.

**Comment by snowtimeglass** [comment]
Thanks. Then, I guess I'd better leave it to Lint Debug to check if the string isn't somewhere else.

**Comment by snowtimeglass** [comment]
After `Bury note` action is done, a message, for example, `2 cards buried.`, is shown on Anki Desktop.
image

In order to reproduce it on AnkiDroid with Anki Desktop strings, what should be done? Filling in some variable in the following parentheses?  

```
internal fun buryNote(): Boolean {
        return dismiss(BuryNote(currentCard!!)) {
            showSnackbarWithUndoButtonText(TR.studyingCardsBuried(●))
```
image

**Comment by BrayanDSO** [comment]
On methods that operate on a note, you can use `currentCard!!.note().numberOfCards()`. If it operates on a card, passing `1` should be fine

**Comment by snowtimeglass** [comment]
I see. Thanks a lot!

**Comment by snowtimeglass** [comment]
@dae On Anki Desktop, the actions of Bury and Suspend produce messages whose styles are different from each other as follows. Is it intentional?

**Action         -->  Message**

`Bury Card` --> `1 card buried.`
`Bury Note` --> `2 cards buried.` (the case of a note with 2 cards)

`Suspend Card` --> `Card suspended.`
`Suspend Note` --> `Note suspended.`

**Comment by mikehardy** [comment]
@dae good question for you just above :point_up:

**Comment by dae** [comment]
Sorry, lost track of this. I think the difference is just because they were added at different times by different people. Suspend card/note came earlier and is only used in the review screen; the bury message was added later, and including the number was likely because it can be used in both the reviewer and browse screen (even though the browse screen doesn't currently provide a bury option, it may in the future).

**Comment by snowtimeglass** [comment]
@dae @mikehardy thanks. Then, I suppose the following hybrid style might be good.

**Action         -->  Message**

`Bury Card` --> `Card buried.`
`Bury Note` --> `n cards buried.` (n≧2) ; `1 card buried.` (the case of executing `Bury Note` gesture on a card which has no siblings. On AnkiDroid, `Bury Note` item on action menu isn't shown for a card with no siblings)

Same for Suspend

`Suspend Card` --> `Card suspended.`
`Suspend Note` --> `n cards suspended.` (n≧2) ; `1 card suspended.` (the case of executing `Suspend Note` gesture on a card which has no siblings. On AnkiDroid, `Suspend Note` item on action menu isn't shown for a card with no siblings)

May I continue working in this direction?

**Comment by BrayanDSO** [comment]
Fine with me

**Comment by snowtimeglass** [comment]
The string of `Card buried` seems to exist only in AnkiMobile resource.
image

What approach would be preferable? 
Using the string of AnkiMobile in some way? 
Requesting creation of that string in Anki Desktop resource? 
Just replacing 
`<string name="buried_card">Buried card</string>` 
with 
`<string name="buried_card">Card buried</string>` in AnkiDroid/src/main/res/values/02-strings.xml? 
Or replacing it with 
`<string name="Card_buried">Card buried</string>`?

**Comment by BrayanDSO** [comment]
If we can use the mobile translation with `CollectionManager.TR`, use it

else if we have an AnkiDroid string equivalent to `Card buried/Buried card`, use it

else, create a new one on AnkiDroid

**Comment by dae** [comment]
The AnkiMobile translations are not easily usable. It's possible to migrate individual translations from mobile→desktop, but it's currently it's a bit of a pain to do so I'm afraid.

**Comment by snowtimeglass** [comment]
Thank you, two. Then, rewriting the following related string on AnkiDroid is currently a handy way, I guess.

`<string name="buried_card">Buried card</string>`

**Comment by snowtimeglass** [comment]
I've made strings (on 02-strings.xml in my local repository) for the snackbar message after the action of "Suspend note".
```
<!-- Reviewer actions -->
    <string name="card_buried">Card buried.</string>
    <plurals name="note_suspended">
        <item quantity="one">%d card suspended</item>
        <item quantity="other">%d cards suspended</item>
    </plurals>
```


Could anyone please advise me how to reflect the number of cards (`%d`) in the following code of AbstractFlashcardViewer.kt to show it on the snackbar message? Using `getQuantityString` and `currentCard!!.note().numberOfCards()`? (Currently, tapping "Suspend note" causes a crash on my debug.apk...)
```
    internal fun buryCard(): Boolean {
        return dismiss(BuryCard(currentCard!!)) {
            showSnackbarWithUndoButton(R.string.card_buried)
        }
    }

    internal fun buryNote(): Boolean {
        return dismiss(BuryNote(currentCard!!)) {
            showSnackbarWithUndoButtonText(TR.studyingCardsBuried(currentCard!!.note().numberOfCards()))
        }
    }

    internal fun suspendCard(): Boolean {
        return dismiss(SuspendCard(currentCard!!)) {
            showSnackbarWithUndoButtonText(TR.studyingCardSuspended())
        }
    }

    internal fun suspendNote(): Boolean {
        return dismiss(SuspendNote(currentCard!!)) {
            showSnackbarWithUndoButton(R.plurals.note_suspended)
        }
    }

```

**Comment by snowtimeglass** [comment]
The results of the last commits are as follows. I would deeply appreciate it if you would give me a review and some advice.
image
image
image
image

File: AnkiDroid/src/main/res/values/02-strings.xml [hunk 1]
```diff
@@ -376,12 +376,13 @@
...
-    <string name="deleted_note">Deleted note</string>
-    <string name="suspended_card">Suspended card</string>
-    <string name="suspended_note">Suspended note</string>
+    <!-- Reviewer actions 
+    Each of these strings ends with a priod, following the notation of Anki Desktop -->
+    <string name="card_buried">Card buried.</string>
+    <plurals name="note_suspended"
+        <item quantity="one">%d card suspended.</item>
+        <item quantity="other">%d cards suspended.</item>
+    </plurals>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/02-strings.xml:385]: Follow up to lint these

**Review by david-allison — APPROVED** [review]
Looks good, I'm going to rebase this one on `main`, as it's been stale for a while and some strings need fixing

**Comment by snowtimeglass** [comment]
Thank you so much for taking the trouble to do the review and the works!

**Review by lukstbit — APPROVED** [review]
LGTM, thanks!

**Comment by mikehardy** [comment]
I'm clearing strings out now then will merge the strings-related stack of PRs queued up

**Pull request merged**